weaver/processes/wps_package.py

```diff
@@ -2107,7 +2112,8 @@ def _handler(self, request, response):
             raise
         else:
             self.update_status("Package complete.", PACKAGE_PROGRESS_DONE, STATUS_SUCCEEDED)
-        return self.response
+        finally:
+            return self.response
```

Comment:
why the finally here?

Reply:
In order to let the handler finish even if it raises an exception. Otherwise, lets say we have a process which fails, "Completed PermanentFailed" is returned but the the execution is stopped here. We want the error to be sent to logs but not to hang the execution since there was an issue.

Reply:
Will re-evaluate when travis re-runs everything after linting is fixed. 
Not sure if not raising the exception correctly updates the response as failed with the traceback for debuging...

Reply:
Yes, you're right. I got this after linting : "return statement in finally block may swallow exception (lost-exception)". I however don't have a proposition, since I think we want handler to raise, but correctly finish, to avoid hanging up the execution..

Reply:
I think that since each raise includes a status_update call beforehand with the appropriate message and job-status value, there is no need for anything else. PyWPS then catches the exceptions and logs everything as needed.